build_tools/utils.py

```diff
@@ -262,15 +261,15 @@ def copy_common_headers(te_src, dst):
         shutil.copy(file_path, new_path)
 
 
-def install_and_import(package):
-    """Install a package via pip (if not already installed) and import into globals."""
-    main_package = package.split("[")[0]
-    subprocess.check_call([sys.executable, "-m", "pip", "install", package])
-    globals()[main_package] = importlib.import_module(main_package)
+def install_packages(packages):
+    """Install a package via pip (if not already installed)."""
+    for package in packages:
+        main_package = package.split("[")[0]
+    subprocess.run([sys.executable, "-m", "pip", "install", package])
```

Comment:
A few issues:
- Only the last package is installed
- The packages need to be available after installed, e.g. at https://github.com/NVIDIA/TransformerEngine/blob/474053ad305a923cc5341359e9ba369110d8e964/setup.py#L38-L39
- It would be useful to print the error message if Pip fails
```suggestion
def install_packages(packages: Iterable[str]) -> None:
    """Install packages via pip (if not already installed)."""
    install_command = [sys.executable, "-m", "pip", "install"]
    import_names = []
    for package in packages:
        install_command.append(package)
        import_names.append(package.split("[")[0])
    subprocess.check_call(install_command)
    for name in import_names:
        globals()[name] = importlib.import_module(name)
```